modules/phase_field/src/userobjects/DiscreteNucleationInserter.C

```diff
@@ -78,13 +81,18 @@ DiscreteNucleationInserter::threadJoin(const UserObject &y)
 {
   // combine _local_nucleus_list entries from all threads on the current process
   const DiscreteNucleationInserter & uo = static_cast<const DiscreteNucleationInserter &>(y);
-  _local_nucleus_list.insert(_local_nucleus_list.end(), uo._local_nucleus_list.begin(), uo._local_nucleus_list.end());
+  _global_nucleus_list.insert(_global_nucleus_list.end(), uo._local_nucleus_list.begin(), uo._local_nucleus_list.end());
+  _changes_made += uo._changes_made;
 }
 
 void
 DiscreteNucleationInserter::finalize()
 {
-  // here we need to combine all _local_nucleus_list into a the _global_nucleus_list
-  _global_nucleus_list = _local_nucleus_list;
+  // add the _local_nucleus_list of thread zero
+  _global_nucleus_list.insert(_global_nucleus_list.end(), _local_nucleus_list.begin(), _local_nucleus_list.end());
+  // combine _global_nucleus_lists from all MPI ranks
   _communicator.allgather(_global_nucleus_list);
```

Comment:
Idk what is installed on hpcbuild? On the Mac I'm using the redistributable.

On Thu, Aug 6, 2015, 7:38 AM John W. Peterson <notifications@github.com>
wrote:

> In modules/phase_field/src/userobjects/DiscreteNucleationInserter.C
> <https://github.com/idaholab/moose/pull/5513#discussion_r36414430>:
>
> >  }
> >
> >  void
> >  DiscreteNucleationInserter::finalize()
> >  {
> > -  // here we need to combine all _local_nucleus_list into a the _global_nucleus_list
> > -  _global_nucleus_list = _local_nucleus_list;
> > +  // add the _local_nucleus_list of thread zero
> > +  _global_nucleus_list.insert(_global_nucleus_list.end(), _local_nucleus_list.begin(), _local_nucleus_list.end());
> > +  // combine _global_nucleus_lists from all MPI ranks
> >    _communicator.allgather(_global_nucleus_list);
>
> _global_nucleus_list is a std::vector of std::pair<Real, Point>? Yeah,
> there's a good chance one compiler is deducing the wrong template function
> for this or something. Are you using different MPI implementations (openMPI
> vs. mpich) on the two machines?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/5513/files#r36414430>.
>